### PR TITLE
Custom params on generate

### PR DESCRIPTION
**Pull request by chitalian**
Allows the ability to add custom parameters within your OpenAI generate calls
```typescript

export const run = async () => {
  const model = new OpenAI({ temperature: 0 });
  const res = await model.call(
    "What would be a good company name a company that makes colorful socks?",
    undefined,
    {
      headers: {
        "User-Id": "123",
      },
    }
  );
  console.log(res);
};
```

### CODE REVIEW

File: langchain/llms/openai.ts [hunk 1]
```diff
@@ -87,6 +89,8 @@ type TokenUsage = {
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 type Kwargs = Record<string, any>;
 
+export type OpenAIRequestConfig = AxiosRequestConfig;
```
Comment (sullivan-sean) [inline langchain/llms/openai.ts:92]: Wondering if we should nest the axios request config in an object type here to future-proof the third arg of generate for any other things we might want to add in the future, e.g.

```suggestion
export type OpenAIExtraArgs = {
    requestConfig: AxiosRequestConfig
};
```

**Review by sullivan-sean — COMMENTED** [review]
LGTM pending type change here once https://github.com/hwchase17/langchainjs/pull/46 is merged

File: langchain/llms/openai.ts (hunk 1 above)
Comment (chitalian) [reply langchain/llms/openai.ts:92]: ahhh totally agree. We should also make it optional as well

**Comment by nfcampos** [comment]
Note to me: combine #263

**Review by agola11 — APPROVED** [review]

**Pull request merged**